internal/nvmeof/controller/controllerserver.go

```diff
@@ -277,13 +280,18 @@ func validateCreateVolumeRequest(req *csi.CreateVolumeRequest) error {
 	params := req.GetParameters()
 	requiredParams := []string{
 		"subsystemNQN", "nvmeofGatewayAddress", "nvmeofGatewayPort",
-		"listenerIpAddress", "listenerPort", "listenerHostname",
+		"listeners",
```

Comment:
Do you really want to require the listeners to be present in the StorageClass? It would be nicer if the listeners can be configured automatically (detect the available gateways?).

Or, do you want to make the listeners in the StorageClass optional at a later point in time?

Reply:
Ok, thanks!